pkg/cloudprovider/aws/instance.go

```diff
@@ -262,6 +327,22 @@ func (p *InstanceProvider) instanceToNode(instance *ec2.Instance, instanceTypes
 	return nil, fmt.Errorf("unrecognized instance type %s", aws.StringValue(instance.InstanceType))
 }
 
+func (p *InstanceProvider) addToIceCache(overrides *ec2.FleetLaunchTemplateOverrides, capacityType string) {
+	instanceType := aws.StringValue(overrides.InstanceType)
+	zone := aws.StringValue(overrides.AvailabilityZone)
+	cacheTTL := iceCacheSpotTTL
+	if capacityType == v1alpha1.CapacityTypeOnDemand {
+		cacheTTL = iceCacheODTTL
+	}
+	cacheKey := createIceCacheKey(instanceType, capacityType)
+	if zones, exists := p.iceCache.Get(cacheKey); exists {
+		zones.(sets.String).Insert(zone)
+		p.iceCache.Set(cacheKey, zones, cacheTTL)
+	} else {
+		p.iceCache.Set(cacheKey, sets.String{zone: sets.Empty{}}, cacheTTL)
```

Comment:
Don't you need to include the zone in the cache? 
```suggestion
		p.iceCache.Set(cacheKey, sets.String{zone: sets.NewString(zones)}, cacheTTL)
```